### ISSUE TITLE
Properly support IPv4/6 dual stack

### ISSUE THREAD

**Issue by minus7** [teeworlds/teeworlds#1294]
IPv6 support has been added in 0.6, but dual stack is not properly supported. The following issues exist (and respective ways to solve them):
* A game server cannot run on IPv4 and IPv6 at once (the network stack and main loop have to be adapted to two sockets)
* Even if it could, the client (assuming it is dual stack too) could not identify 2 addresses belonging to one server (could be solved by delivering both addresses in the server list tied together, i.e. a server entry has IPv4, IPv6 and port)
* The master server has no way to associate 2 addresses with one server either (would need some kind of server token)
* Dual stack masters don't play well with the current way of splitting/sharding the server list #1292 (same solution as to the second point)

This is one way achieving dual stack support, but I'm not sure it's the best.

**Comment by heinrich5991** [comment]
The first point actually works today. This can be observed on a IPv6-ready host, on the LAN tab.

**Comment by minus7** [comment]
Right, an IPv6 socket usually can send IPv4 too. A second socket is still required when listening on specific addresses though because you need to bind to two addresses then.

**Comment by heinrich5991** [comment]
@minus7 Internally, two sockets are used. The function lacks an interface for binding to specific IPv4/IPv6 addresses though, it only works for the standard (which should be the most deployed anyway).

**Comment by Fisico** [comment]
maybe there is a way to implement dns/reversedns... than dualstack should be less bad to add to the masterserver, servers, and Clients.
